### PR TITLE
Introduce skills controller and CSV processing

### PR DESCRIPTION
**Pull request by elxsy**
Skills controller to upload and display the form

### CODE REVIEW

File: src/controllers/skillsController.ts [hunk 1]
```diff
@@ -82,16 +85,27 @@ export class SkillsController {
...
+
 				const quiz = new Quiz()
-				quiz.profession = profession
+				quiz.profession = policyProfession
 				quiz.questions = questionToSend
 
-				console.log(quiz)
-
-				await this.csrsService.postSkills(quiz).then(() =>{
+				console.log("Quiz: ", quiz)
```
Comment (elxsy) [inline src/controllers/skillsController.ts]: please remove console.log

File: views/page/skills/skills.html [hunk 2]
```diff
@@ -11,7 +11,7 @@ <h1 class="govuk-heading-l">Skills</h1>
         <div class="reports-container">
             <ul class="list">
                 <li class="list-details">
-                    <form action="/content-management/skills" method="post" class="skills-form" enctype="multipart/form-data">
+                    <form action="/content-management/skills" method="post" class="skills-form">
```
Comment (elxsy) [inline views/page/skills/skills.html]: removed the form upload capability, please revert this line to how it was

File: src/controllers/skillsController.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+				req.session!.sessionFlash = {uploadedFileNotCSV}
+				req.session!.save(() => {
+					res.redirect('/content-management/skills')
+				})
+			} else {
+
+				// @ts-ignore
+				await csvtojson().fromString(req.files.file.data.toString('utf-8')).then(async (questions: any) => {
+
+					const opts = ['A', 'B', 'C', 'D', 'E']
```
Comment (m24murray) [inline src/controllers/skillsController.ts]: Do we know for certain that each question will consist of 5 possible answers (A-E)?

File: src/controllers/skillsController.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+	uploadAndProcess() {
+		return async (req: Request, res: Response, next: NextFunction) => {
+			let uploadedFileNotCSV: boolean = false
+
+			// @ts-ignore
+			const uploadedFile = req.files.file
+			let fileExtension = uploadedFile.name.split('.')
+			let fileType = fileExtension[fileExtension.length - 1]
+
+			console.log("fileType: ", fileType)
```
Comment (m24murray) [inline src/controllers/skillsController.ts]: Can you remove this console log

File: views/component/nav/template.njk [hunk 5]
```diff
@@ -11,6 +11,7 @@
             {% if identity.isCshrReporter() or identity.isOrganisationReporter() or identity.isProfessionReporter() or identity.isKPMGSupplierReporter()  or identity.isKornferrySupplierReporter() %}
                 <a class="nav-item govuk-link {% if originalUrl === '/reporting' %}active{% endif %}" href="/reporting">Reporting</a>
             {% endif %}
+            <a class="nav-item govuk-link" href="/skills">Skills</a>
```
Comment (m24murray) [inline views/component/nav/template.njk]: please change to `/content-management/skills`

**Pull request merged**